jwala-tomcat/src/main/resources/data/scripts/thread-dump.sh

```diff
@@ -8,19 +8,38 @@ esac
 export JAVA_HOME=$1
 export JVM_INSTANCE_DIR=$2
 export JVM_NAME=$3
-
-if [ ! -d ${JVM_INSTANCE_DIR}/thread_dump_reports} ]; then
-      mkdir -p ${JVM_INSTANCE_DIR}/thread_dump_reports
-fi
+export DATA_DIR=$4
+export DUMP_FILE=$5
+JWALA_EXIT_CODE_SUCCESS=0
+JWALA_EXIT_CODE_FAILED=1
 
 current_time=$(date +"%m-%d-%Y-%T")
 
 if $linux; then
 	echo $(<${JVM_INSTANCE_DIR}/logs/catalina.pid)
-	/usr/bin/sudo -u tomcat ${JAVA_HOME}/bin/jstack $(<${JVM_INSTANCE_DIR}/logs/catalina.pid) 2>&1 | tee  ${JVM_INSTANCE_DIR}/thread_dump_reports/thread_dump_$current_time
+	thread_dump_output=$(/usr/bin/sudo -u tomcat ${JAVA_HOME}/bin/jstack $(<${JVM_INSTANCE_DIR}/logs/catalina.pid) 2>&1 )
+	THREAD_DUMP_RETURN_CODE=$?
+    if echo "$thread_dump_output" | grep -iqF "Full thread dump"; then
+	   echo "$thread_dump_output" |tee ${DATA_DIR}/$DUMP_FILE
+	   echo "Creating thread-dump file at location: ${DATA_DIR}/$DUMP_FILE, "
+	else
+	   echo "Generation of thread dump failed : $thread_dump_output"
+	fi
+	exit $THREAD_DUMP_RETURN_CODE
 fi
 
 if $cygwin; then
     export JVMPID=`sc queryex $JVM_NAME | /usr/bin/grep PID | /usr/bin/awk '{ print $3 }'`
-	${JAVA_HOME}/bin/jstack -l ${JVMPID} 2>&1 |tee ${JVM_INSTANCE_DIR}/thread_dump_reports/thread_dump_$current_time
-fi
\ No newline at end of file
+    thread_dump_output=$(${JAVA_HOME}/bin/jstack -l ${JVMPID} 2>&1 )
+    THREAD_DUMP_RE  TURN_CODE=$?
+    if echo "$thread_dump_output" | grep -iqF "Full thread dump"; then
+       echo "$thread_dump_output" |tee ${DATA_DIR}/$DUMP_FILE
+	   echo "Creating Thread Dump file at location : ${DATA_DIR}/$DUMP_FILE, "
+	   exit $JWALA_EXIT_CODE_SUCCESS
```

Comment:
Why use 2 different ways of returning the status code?
For the linux thread dump the exit status is stored in the THREAD_DUMP_RETURN_CODE, but here for the cygwin way the status code is returned as a constant. Since you check specifically for the "Full thread dump" substring I would use the explicit return of the constant for both ways - that is, change the linux way to return the constant and don't even check for the jstack return value.

Also on line 34, the THREAD_DUMP_RETURN_CODE has a space in the constant name and is never used so get rid of it anyway.